src/Polymorph-Widgets/PharoDarkTheme.class.st

```diff
@@ -285,271 +153,157 @@ PharoDarkTheme >> initialize [
 
 { #category : #'label-styles' }
 PharoDarkTheme >> labelDisabledColorFor: aLabel [
-	"Answer the color to use for a label when disabled."
 
-	^aLabel owner
-		ifNil: [aLabel paneColor muchDarker]
-		ifNotNil: [aLabel owner color isTransparent
-			ifTrue: [ aLabel owner paneColor muchLighter ]
-			ifFalse: [ aLabel owner color muchLighter ] ]
-]
-
-{ #category : #'accessing - colors' }
-PharoDarkTheme >> lessConspicuousColorFrom: aColor [
-
-	^	aColor slightlyLighter
-]
-
-{ #category : #'accessing - colors' }
-PharoDarkTheme >> lightBackgroundColor [
-
-	^ Color r: 0.218 g: 0.217 b: 0.217
-]
-
-{ #category : #'accessing - colors' }
-PharoDarkTheme >> lineNumberColor [ 
-	^ Color gray lighter lighter
-]
-
-{ #category : #'fill-styles' }
-PharoDarkTheme >> listNormalFillStyleFor: aList [
-	"Return the normal fillStyle for the given list."
-	
-	^ self backgroundColor
-]
-
-{ #category : #'accessing - colors' }
-PharoDarkTheme >> listTextColor [
-	"Answer the list text color to use."
-
-	^ self textColor
-]
+	"Answer the color to use for a label when disabled."
 
-{ #category : #'accessing - colors' }
-PharoDarkTheme >> menuBorderColor [
-	^ self borderColor
+	^ aLabel owner ifNil: [ aLabel paneColor muchDarker ] ifNotNil: [ 
+		  aLabel owner color isTransparent ifTrue: [ aLabel owner paneColor muchLighter ]
+			  ifFalse: [ aLabel owner color muchLighter ] ]
```

Comment:
Color allocationS here!

Reply:
For this one, I don't know how to cache it. Do you think is a good idea to have 3 methods. For example: `disabledLabelWhenOwnerIsNil` `disabledLabel` `disabledLabelWhenTransparent` ?